src/components/GraphFilter/GraphFilter.tsx

```diff
@@ -74,10 +77,12 @@ export default class GraphFilter extends React.Component<GraphFilterProps, Graph
             disabled={this.props.disabled}
             onClick={this.updateDuration}
             nameDropdown={'Duration'}
-            initialValue={this.props.graphDuration.value}
+            initialValue={Number(sessionStorage.getItem('appDuration')) || this.props.graphDuration.value}
```

Comment:
Number() is a built-in conversion helper. isn't it

Reply:
Yes, the result of getItem from the storage is a string and I need to cast to number.

Reply:
@aljesusg  Similar logic for restoring params from local storage is in `ServiceGraphRouteHandler`.   May I suggest that `appDuration` is initialized there as well and is passed down to child components as prop instead?

this may be useful: https://github.com/kiali/kiali-ui/pull/243#issuecomment-383694057

Reply:
Tolbar metrics need appDuration too, we shouldn't pass this like props when in each toolbar is setting  a new appDuration in sessionStorage when user select from dropdown.